containers/src/Kokkos_Vector.hpp

```diff
@@ -138,6 +137,7 @@ class KOKKOS_DEPRECATED vector
 
     DV::view_host()(_size) = val;
     _size++;
+    DV::template modify<typename DV::t_host::device_type>();
```

Comment:
Why this move? Ah I guess in case of resize?

Reply:
Yes, this is to fix the test we introduced for https://github.com/kokkos/kokkos/issues/5042.